Content.Server/Chat/ChatCommands.cs

```diff
@@ -72,4 +78,81 @@ public void Execute(IConsoleShell shell, IPlayerSession player, string[] args)
             chat.SendOOC(player, string.Join(" ", args));
         }
     }
+
+    internal class SuicideCommand : IClientCommand
+    {
+        public string Command => "suicide";
+
+        public string Description => "Commits suicide";
+
+        public string Help => "suicide";
+
+        private void DealDamage(ISuicideAct suicide, IChatManager chat, DamageableComponent damageableComponent, IEntity source, IEntity target)
+        {
+            SuicideKind kind = suicide.Suicide(target, chat);
+            if (kind != SuicideKind.Special && damageableComponent != null)
+            {
+                damageableComponent.TakeDamage(kind switch
+                {
+                    SuicideKind.Brute    => DamageType.Brute,
+                    SuicideKind.Heat     => DamageType.Heat,
+                    SuicideKind.Cold     => DamageType.Cold,
+                    SuicideKind.Acid     => DamageType.Acid,
+                    SuicideKind.Toxic    => DamageType.Toxic,
+                    SuicideKind.Electric => DamageType.Electric,
+                                       _ => DamageType.Brute
+                },
+                500, //TODO: needs to be a max damage of some sorts
+                source,
+                target);
+            }
+        }
+
+        public void Execute(IConsoleShell shell, IPlayerSession player, string[] args)
+        {
+            if (player.Status != SessionStatus.InGame)
+                return;
+
+            var chat = IoCManager.Resolve<IChatManager>();
+            var owner = player.ContentData().Mind.OwnedMob.Owner;
+            owner.TryGetComponent<DamageableComponent>(out var dmgComponent);
+            //TODO: needs to check if the mob is actually alive
+            //TODO: maybe set a suicided flag to prevent ressurection?
+
+            // Held item suicide
+            owner.TryGetComponent<HandsComponent>(out var handsComponent);
```

Comment:
You're not quite using `TryGetComponent` correctly. If the component should always be there, then you should use `GetComponent`. If the component can be missing, then you should check the return value of `TryGetComponent` - if it's `true`, then the `out var` will never be `null`.

For example, you use `TryGetComponent<HandsComponent>`, but then you immediately use `handsComponent` as if it were not `null` - so you can replace that with `GetComponent`.

For the `DamageableComponent`, you can decide for yourself if it is allowed to be missing. But if it is, you should wrap all this code in a `if (owner.TryGetComponent<DamageableComponent>(out var dmgComponent) { <your code > }` and remove the `null`-checks for `dmgComponent`, since it will not be `null` inside that if-statement.